onetun.h

```diff
@@ -0,0 +1,40 @@
+#include <stdarg.h>
```

Comment:
Was this generated with `cbindgen`? Would I need to run a command whenever the entrypoint API changes? If so, which command?

Any way to move this and `cbindgen.toml` under its own subdirectory, `/ffi`? I'm also considering moving this under a new crate `onetun-ffi`, if that's the convention.

Reply:
Yes, I generated them with ``cbindgen``. When you update the ``ffi.rs`` file, just run ``cbindgen --config cbindgen.toml --crate onetun --output onetun.h`` and it'll recreate the header file.

I can definitely move the cbindgen.toml file, would you like the header moved as well? As far as convention goes, the extern functions are usually left within the same crate, as can be seen [here](https://github.com/cloudflare/boringtun/blob/2fffe3c763513ace6f30e7bb35e90b940be84c64/boringtun/src/ffi/mod.rs) with boringtun. If you want to move it anyways, if you create the repository I can open this pull request there.

Reply:
I'm just thinking it might make sense to automatically generate this file every release instead of adding it in the source code and manually updating it (I tend to be forgetful). It could be included as a GitHub release asset, as well as the various library files (`.so`, `.a`, and so on).

You don't have to include that in your PR, but curious if that would still fit your needs?

Reply:
The problem about not adding this file in the source code is that people who use this library will have to generate it themselves, which is not really the convention. If this file isn't part of the repository, maybe it could be automatically generated with a ``build.rs``?

As for uploading the libraries, I'm not familiar with GitHub actions and it would be limited to the architectures and OS's GitHub supports.

Reply:
Yes `build.rs` would be the preferred approach in this case. There is a good example in the `cbindgen` docs: https://github.com/eqrion/cbindgen/blob/master/docs.md#buildrs

IMO the output path should then be `target/onetun.h`.

Thanks for your help so far. Let me know if you can implement this change as well.